tests/e2e/v2store_deprecation_test.go

```diff
@@ -56,7 +56,7 @@ func assertVerifyCanStartV2deprecationNotYet(t testing.TB, dataDirPath string) {
 
 	if err := e2e.CURLGet(epc, e2e.CURLReq{
 		Endpoint: "/v2/keys/foo",
-		Expected: `{"action":"get","node":{"key":"/foo","value":"bar9","modifiedIndex":13,"createdIndex":13}}`}); err != nil {
+		Expected: `{"action":"get","node":{"key":"/foo","value":"bar9","modifiedIndex":14,"createdIndex":14}}`}); err != nil {
```

Comment:
Would like to better understand the source of the changes in index. If possible separate changes in logic from  changes to tests.

Reply:
caused by https://github.com/etcd-io/etcd/pull/19563#discussion_r1988974045